### ISSUE TITLE
[bug 🐛] incorrect number of tokens to copy

### ISSUE THREAD

**Issue by Niccolo-Ajroldi** [athms/mad-lab#7]
**tl;dr**: there is a bug in the data generation pipeline for the **Selective Copying** task: the generated samples always contain **16** tokens to be copied, regardless of the value of `num_tokens_to_copy` specified by the user.
## Description

Data are generated in `train.py` from: 
```python
data = generate_data(
    instance_fn=mad_config.instance_fn,
    instance_fn_kwargs=mad_config.instance_fn_kwargs,
)
```
The bug is in the implementation of the [`instance_fn_kwargs`](https://github.com/athms/mad-lab/blob/69d09e29c1b4b642a1f468fc935fb8ccd366204c/mad/configs.py#L81) property of `mad_config`, which is not adding `num_tokens_to_copy` to the returned dictionary:

```python
    @property
    def instance_fn_kwargs(self) -> tp.Dict:
        """returns dict of all kwargs required to create an instance with self.instance_fn"""
        ...
        return dict(
            vocab_size=self.vocab_size,
            seq_len=self.seq_len,
            k_motif_size=self.k_motif_size,
            v_motif_size=self.v_motif_size,
            frac_noise=self.frac_noise,
            noise_vocab_size=self.noise_vocab_size,
            num_tokens_to_copy=self.num_tokens_to_copy,  # <-- THIS LINE IS MISSING IN THE ORIGINAL CODE!
            rng=np.random.default_rng(self.seed),
            multi_query=self.multi_query,
            kv_map=kv_map
        )

```

Without this line, the data generating function `mad_config.instance_fn` always resorts to the default value of `num_tokens_to_copy=16`, resulting in incorrect generated data.

## FIx
Pull-request #8 adds the aforementioned line and fixes this bug.

**Comment by Niccolo-Ajroldi** [comment]
any update @Zymrael ?
